cmd/contour/serve.go

```diff
@@ -638,30 +638,20 @@ func (s *Server) doServe() error {
 	return s.mgr.Start(signals.SetupSignalHandler())
 }
 
-func (s *Server) setupTracingService(tracingConfig *contour_api_v1alpha1.TracingConfig) (*xdscache_v3.TracingConfig, error) {
-	if tracingConfig == nil {
-		return nil, nil
-	}
-
-	// ensure the specified ExtensionService exists
+func getExtensionSvcProperties(name string, namespace string) (*xdscache_v3.ExtensionServiceConfig, error) {
 	extensionSvc := &contour_api_v1alpha1.ExtensionService{}
 	key := client.ObjectKey{
-		Namespace: tracingConfig.ExtensionService.Namespace,
-		Name:      tracingConfig.ExtensionService.Name,
+		Namespace: namespace,
+		Name:      name,
 	}
-	// Using GetAPIReader() here because the manager's caches won't be started yet,
-	// so reads from the manager's client (which uses the caches for reads) will fail.
-	if err := s.mgr.GetAPIReader().Get(context.Background(), key, extensionSvc); err != nil {
-		return nil, fmt.Errorf("error getting tracing extension service %s: %v", key, err)
-	}
```

Comment:
This block of code needs to be included in `getExtensionSvcProperties` as well. You could either pass in the APIReader as an argument for the function to use, or change `getExtensionSvcProperties` to be a method on `s` so that it can access `s.mgr` directly. I'd be fine with either approach.